src/createServer.js

```diff
@@ -1,3 +1,40 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { validateUrl } = require('./validateUrl');
+const { convertToCase } = require('./convertToCase');
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+    const { pathname, searchParams } = normalizedUrl;
+
+    const text = pathname.slice(1);
+    const toCase = searchParams.get('toCase');
+
+    const errors = validateUrl(text, toCase);
+
+    if (errors.length) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad Request';
+      res.setHeader('Content-Type', 'application/json');
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const convertedData = convertToCase(text, toCase);
+
+    res.statusCode = 200;
```

Comment:
You have duplicated code here and on 16-19 lines. We can create a function with arguments to avoid duplication. That function just receives status code, message and data for JSON.stringify into arguments. It will be easily to use and reuse this function in future.